Core/src/org/sleuthkit/autopsy/communications/VisualizationPanel.java

```diff
@@ -674,7 +715,116 @@ private void fitGraph() {
 
         graphComponent.zoom((heightFactor + widthFactor) / 2.0);
     }
+    
+    /**
+     * Handle the ActionPerformed event from the Snapshot button.
+     * 
+     * @throws NoCurrentCaseException
+     * @throws IOException 
+     */
+    @NbBundle.Messages({
+        "VisualizationPanel_action_dialogs_title=Communications",
+        "VisualizationPanel_action_name_text=Snapshot Report",
+        "VisualizationPane_fileName_prompt=Enter name for the Communications Snapshot Report:",
+        "VisualizationPane_reportName=Communications Snapshot",})
+    private void handleSnapshotEvent() throws NoCurrentCaseException, IOException {
+        Case currentCase = Case.getCurrentCaseThrows();
+        Date generationDate = new Date();
+
+        final String defaultReportName = FileUtil.escapeFileName(currentCase.getDisplayName() + " " + new SimpleDateFormat("MMddyyyyHHmmss").format(generationDate)); //NON_NLS
+
+        final JTextField text = new JTextField(50);
+        final JPanel panel = new JPanel(new GridLayout(2, 1));
+        panel.add(new JLabel(Bundle.VisualizationPane_fileName_prompt()));
+        panel.add(text);
+
+        text.setText(defaultReportName);
+
+        int result = JOptionPane.showConfirmDialog(graphComponent, panel,
```

Comment:
can you use JOptionPane.showInoutDialog() here, and eliminate the  GUI config code?

Reply:
There is no simple way to get both the value of a text field and determine which of the buttons was clicked.  If you want the behavior of assuming that empty means default instead of cancel this is the simplest way to achieve that functionality.

Reply:
ok, makes sense